woocommerce/woocommerce-gutenberg-products-block
# Create shared contexts handle for inner blocks

_#2399 is a mammoth PR so I'm breaking out some fixes and changes to review separately._

Background behind these changes can be seen in this comment from @nerrad https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2399#issuecomment-631560641

Changes:

- Shared contexts so Atomic blocks are functional (see #2399 (comment)) This ensures contexts work when wrapping atomic blocks in the editor and on frontend.
- Combines useProductLayoutContext and useInnerBlockConfigurationContext since both are used at the same time in all contexts. This allows the parent block to be passed, as well as a prefix for the CSS classnames.
- Introduces ProductDataContext to share product data with atomic blocks. This is not in use yet; it is for the single product block.

### How to test the changes in this Pull Request:

This isn't in use by multiple blocks yet, so to test please just Smoke Test the All Products block and ensure there are no regressions in the layout (front and backend).
